flutter_ffi_plugin/bin/rinf_config.dart

```diff
@@ -0,0 +1,120 @@
+import 'dart:io';
+
+import 'package:yaml/yaml.dart';
+
+class RinfConfigMessage {
+  final String inputDir;
+  final String rustOutputDir;
+  final String dartOutputDir;
+
+  RinfConfigMessage._({
+    required this.inputDir,
+    required this.rustOutputDir,
+    required this.dartOutputDir,
+  });
+
+  factory RinfConfigMessage.defaultConfig() {
+    return RinfConfigMessage._(
+      inputDir: DEFAULT_INPUT_DIR,
+      rustOutputDir: DEFAULT_RUST_OUTPUT_DIR,
+      dartOutputDir: DEFAULT_DART_OUTPUT_DIR,
+    );
+  }
+
+  factory RinfConfigMessage.from(YamlMap yaml) {
+    for (final key in yaml.keys) {
+      if (!MESSAGE_CONFIG_KEYS.contains(key)) {
+        throw Exception(
+          "Unknown key '$key' in rinf message configuration.\n"
+          "Available keys are: $MESSAGE_CONFIG_KEYS",
+        );
+      }
+    }
+    return RinfConfigMessage._(
+      inputDir: yaml[KEY_INPUT_DIR] ?? DEFAULT_INPUT_DIR,
+      rustOutputDir: yaml[KEY_RUST_OUTPUT_DIR] ?? DEFAULT_RUST_OUTPUT_DIR,
+      dartOutputDir: yaml[KEY_DART_OUTPUT_DIR] ?? DEFAULT_DART_OUTPUT_DIR,
+    );
+  }
+
+  @override
+  String toString() {
+    return '''message:
+    $KEY_INPUT_DIR: $inputDir
+    $KEY_RUST_OUTPUT_DIR: $rustOutputDir
+    $KEY_DART_OUTPUT_DIR: $dartOutputDir''';
+  }
+
+  static const KEY_INPUT_DIR = "input_dir";
+  static const KEY_RUST_OUTPUT_DIR = "rust_output_dir";
+  static const KEY_DART_OUTPUT_DIR = "dart_output_dir";
+
+  static const DEFAULT_INPUT_DIR = "messages";
+  static const DEFAULT_RUST_OUTPUT_DIR = "native/hub/src/messages";
+  static const DEFAULT_DART_OUTPUT_DIR = "lib/messages";
```

Comment:
My assumption is that we might need to add slash at the back or the front to make it align with all other code, but not sure right now. Maybe currently, the proto file can be generated like `lib/messagescounter_number.proto`. I should need to investigate more.